modules/statistics/php/stats_behavioural.class.inc

```diff
@@ -157,7 +172,10 @@ class Stats_Behavioural extends \NDB_Form
                     AND s.Current_stage <> 'Recycling Bin'
                     AND f.CommentID NOT LIKE 'DDE%'
                     AND s.CenterID <> '1'
-	            AND s.CenterID IN (" . $sitesString . ")
+                    AND s.CenterID IN
+                        (" . implode(',', $paramCenters) . ")
+                    AND c.RegistrationProjectID IN
```

Comment:
shouldn't this be `s.ProjectID`? (@ridz1208, maybe you have input?)

This also looks like it'll break with an SQL error if the `$paramCenters` or `$paramProjects` is empty, but the old code looked like that would happen (for sites at least) too

Reply:
regarding to the use `s.ProjectID` or `c.RegistrationProjectID` my knowledge of this part of the business is not the best, but session (s) makes more sense to me right now so `s.ProjectID` should by the correct one.

regarding to the possibility of a SQL break, while not to much probable It is possible and this is an error in the code which it not taking this alternative into consideration. In this case we're talking about a user(logged user) that doesn't have any project assigned (and the analogous for the sites) [while mandatory during the user creation it could happen, or not... not sure.. but agree it is not a good code]: In this case I think the correct should be to show some kind of message alerting there is not data accessible to the user and the reason (not belonging to any project (or site). ?

Reply:
@driusan, @ridz1208:

**IMPORTANT:**  Before this fixes the module was showing 500 error. So this could be critical to release. While `s.ProjectID `or `c.RegistrationProjectID` is a simple change, the SQL issue could be a bit more elaborated. Again this is not to probably to happen... but no sure if there is cases it could. What do you think we should do in any case?